### PR TITLE
Add ulx propify and ulx cheese

### PR DESCRIPTION
**Pull request by legokidlogan**

### CODE REVIEW

File: lua/ulx/modules/sh/cfc_propify.lua [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    elseif keyNum == IN_MOVERIGHT then
+        local phys = ply.ragdoll:GetPhysicsObject()
+        phys:ApplyForceCenter( ply:EyeAngles():Right() * HOP_STRENGTH * phys:GetMass() )
+    elseif keyNum == IN_MOVELEFT then
+        local phys = ply.ragdoll:GetPhysicsObject()
+        phys:ApplyForceCenter( -ply:EyeAngles():Right() * HOP_STRENGTH * phys:GetMass() )
+    elseif keyNum == IN_JUMP then
+        local phys = ply.ragdoll:GetPhysicsObject()
+        phys:ApplyForceCenter( Vector( 0, 0, HOP_STRENGTH * phys:GetMass() ) )
+    end
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_propify.lua:196]: - set `ply:EyeAngles()` to a local variable
- set `HOP_STRENGTH * phys:GetMass()` as a local variable as well
- phys should just be defined once before the if elseif chain

so it would become something like
```lua
if keyNum == IN_FORWARD then
    phys:ApplyForceCenter( eyeAngles:Forward() * hopStrength )
elseif ... etc
```

File: lua/ulx/modules/sh/cfc_propify.lua [hunk 2]
```diff
@@ -0,0 +1,218 @@
+CFCUlxCommands.propify = CFCUlxCommands.propify or {}
+local cmd = CFCUlxCommands.propify
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_propify.lua:2]: I think this should be `CFCUlxCommands.Propify` with a capital P instead

File: lua/ulx/modules/sh/cfc_cheese.lua [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "models/props_c17/playgroundtick-tack-toe_block01a.mdl"
+}
+
+function cmd.cheese( caller, targets, shouldUncheese )
+    local prop = CFCUlxCommands.propify.propify( caller, targets, BASE_CHEESE_MODELS[ math.random( #BASE_CHEESE_MODELS ) ], shouldUncheese )
+    
+    if not prop then return end
+    
+    prop:SetMaterial( "models/XQM/Rails/gumball_1" )
+    prop:SetColor( Color( 255, math.Rand( 150, 221 ), 6, 255 )  )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_cheese.lua:18]: Try to break this out into a few steps, try to avoid nested calls like this

File: lua/ulx/modules/sh/cfc_cheese.lua [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+local CATEGORY_NAME = "Fun"
+local BASE_CHEESE_MODELS = {
+    "models/hunter/triangles/025x025.mdl",
+    "models/hunter/triangles/05x05x05.mdl",
+    "models/hunter/triangles/1x05x05.mdl",
+    "models/props_c17/playgroundtick-tack-toe_block01a.mdl"
+}
+
+function cmd.cheese( caller, targets, shouldUncheese )
+    local prop = CFCUlxCommands.propify.propify( caller, targets, BASE_CHEESE_MODELS[ math.random( #BASE_CHEESE_MODELS ) ], shouldUncheese )
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_cheese.lua:13]: If you move the cheese model into its own var above this one, this line will be very simple.
```lua
local model = BASE_CHEESE_MODELS[ math.blah
local prop = CFCUlxCommands.Propify.propify( caller, targets, cheeseModel, shouldUncheese )
```

File: lua/ulx/modules/sh/cfc_cheese.lua [hunk 5]
```diff
@@ -0,0 +1,27 @@
+CFCUlxCommands.cheese = CFCUlxCommands.cheese or {}
+local cmd = CFCUlxCommands.cheese
+
+local CATEGORY_NAME = "Fun"
+local BASE_CHEESE_MODELS = {
+    "models/hunter/triangles/025x025.mdl",
+    "models/hunter/triangles/05x05x05.mdl",
+    "models/hunter/triangles/1x05x05.mdl",
+    "models/props_c17/playgroundtick-tack-toe_block01a.mdl"
+}
```
Comment (brandonsturgeon) [inline lua/ulx/modules/sh/cfc_cheese.lua:10]: These could go in `cmd` too

**Review by brandonsturgeon — CHANGES_REQUESTED** [review]

File: lua/ulx/modules/sh/cfc_cheese.lua [hunk 6]
```diff
@@ -0,0 +1,29 @@
+CFCUlxCommands.Cheese = CFCUlxCommands.Cheese or {}
+local cmd = CFCUlxCommands.Cheese
```
Comment (samuelWilliams99) [inline lua/ulx/modules/sh/cfc_cheese.lua:2]: All the other commands have the subtable in camel case, if we wanna take that, it should be it's own PR.
Keep it camel for now
```suggestion
CFCUlxCommands.cheese = CFCUlxCommands.cheese or {}
local cmd = CFCUlxCommands.cheese
```

File: lua/ulx/modules/sh/cfc_propify.lua [hunk 7]
```diff
@@ -0,0 +1,208 @@
+CFCUlxCommands.Propify = CFCUlxCommands.Propify or {}
+local cmd = CFCUlxCommands.Propify
```
Comment (samuelWilliams99) [inline lua/ulx/modules/sh/cfc_propify.lua:2]: Same reason as before
```suggestion
CFCUlxCommands.propify = CFCUlxCommands.propify or {}
local cmd = CFCUlxCommands.propify
```

**Review by samuelWilliams99 — CHANGES_REQUESTED** [review]

File: lua/ulx/modules/sh/cfc_propify.lua [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+local function disallowGrab( ply, _ )
+    if ply.ragdoll then return false end
+    return true
+end
+hook.Add( "AllowPlayerPickup", "CFC_ULXPropifyDisallowGrab", disallowGrab )
+
+--Prevents breakable props from existing after being broken
+local function removePropOnBreak( _, prop )
+    if not prop.ragdolledPly then return end
+    cmd.propifyTargets( __, { prop.ragdolledPly }, __, true )
```
Comment (plally) [inline lua/ulx/modules/sh/cfc_propify.lua:201]: I dont like the double underscores, perhaps you could just use `nil`

**Review by plally — CHANGES_REQUESTED** [review]

**Review by brandonsturgeon — APPROVED** [review]

**Review by brandonsturgeon — APPROVED** [review]
